java/BUILD.bazel

```
@@ -264,6 +264,12 @@ genrule(
         WORK_DIR="$$(pwd)"
         rm -rf "$$WORK_DIR/python/ray/jars" && mkdir -p "$$WORK_DIR/python/ray/jars"
         cp -f $(location //java:ray_dist_deploy.jar) "$$WORK_DIR/python/ray/jars/ray_dist.jar"
+        chmod +w "$$WORK_DIR/python/ray/jars/ray_dist.jar"
+        zip -d "$$WORK_DIR/python/ray/jars/ray_dist.jar" native/*/gcs_server
+        zip -d "$$WORK_DIR/python/ray/jars/ray_dist.jar" native/*/libray_redis_module.so
+        zip -d "$$WORK_DIR/python/ray/jars/ray_dist.jar" native/*/plasma_store_server
+        zip -d "$$WORK_DIR/python/ray/jars/ray_dist.jar" native/*/raylet
+        zip -d "$$WORK_DIR/python/ray/jars/ray_dist.jar" native/*/redis-server
```

Would be nicer to avoid repeating the file name:
```suggestion
        zip -d "$$WORK_DIR/python/ray/jars/ray_dist.jar" \
            native/*/gcs_server \
            native/*/libray_redis_module.so \
            native/*/plasma_store_server \
            native/*/raylet \
            native/*/redis-server
        chmod +w "$$WORK_DIR/python/ray/jars/ray_dist.jar"
```

---

That's great, thanks